packages/ui/src/data-table/data-table.tsx

```diff
@@ -240,124 +275,137 @@ export function DataTable<TData, TValue>({
 
   return (
     <>
-      <div className={cn(`overflow-hidden rounded-lg border bg-background-secondary`, wrapperClass)}>
-        {(showFilter || showVisibility) && (
-          <div className="flex items-center py-4">
-            {showFilter && (
-              <Input
-                placeholder="Filter by name..."
-                value={(table.getColumn('name')?.getFilterValue() as string) ?? ''}
-                onChange={(event) => table.getColumn('name')?.setFilterValue(event.target.value)}
-                className="max-w-sm"
+      <div className="relative">
+        <div className={cn(`overflow-hidden rounded-lg border bg-background-secondary`, wrapperClass)}>
+          {(showFilter || showVisibility) && (
+            <div className="flex items-center py-4">
+              {showFilter && (
+                <Input
+                  placeholder="Filter by name..."
+                  value={(table.getColumn('name')?.getFilterValue() as string) ?? ''}
+                  onChange={(event) => table.getColumn('name')?.setFilterValue(event.target.value)}
+                  className="max-w-sm"
+                />
+              )}
+              {showVisibility && (
+                <DropdownMenu>
+                  <DropdownMenuTrigger asChild>
+                    <Button icon={<EyeIcon />} iconPosition="left" variant="outline" size="md" className="ml-auto mr-2">
+                      Visibility
+                    </Button>
+                  </DropdownMenuTrigger>
+                  <DropdownMenuContent align="end">
+                    {table
+                      .getAllColumns()
+                      .filter((column) => column.getCanHide())
+                      .map((column, index) => (
+                        <DropdownMenuCheckboxItem key={`${column.id}-${index}`} className="capitalize" checked={column.getIsVisible()} onCheckedChange={(value) => column.toggleVisibility(!!value)}>
+                          {column.id}
+                        </DropdownMenuCheckboxItem>
+                      ))}
+                  </DropdownMenuContent>
+                </DropdownMenu>
+              )}
+            </div>
+          )}
+
+          {/* Apply opacity and disable interactions while loading */}
+          <div className={cn(isLoading ? 'opacity-50 pointer-events-none transition-opacity duration-300' : 'transition-opacity duration-300', 'relative overflow-x-auto')}>
+            {showLeftFade && (
+              <div
+                className="absolute left-0 top-5 w-8 bg-gradient-to-r from-border to-transparent pointer-events-none z-10"
```

Comment:
Let's try a little more subtle
```suggestion
                className="absolute left-0 top-5 w-6 bg-gradient-to-r from-border to-transparent pointer-events-none z-10"
```